docs/clients/promtail/configuration.md

```diff
@@ -580,6 +585,68 @@ labels:
 [path: <string>]
 ```
 
+### syslog_config
+
+The `syslog_config` block configures a syslog listener allowing users to push
+logs to promtail with the syslog protocol.
+Currently supported is [IETF Syslog (RFC5424)](https://tools.ietf.org/html/rfc5424)
+with and without octet counting.
+
+The recommended deployment is to have a dedicated syslog forwarder like **syslog-ng** or **rsyslog**
+in front of promtail. The forwarder can take care of the various specifications
+and transports that exist (UDP, BSD syslog, ...).
+
+[Octet counting](https://tools.ietf.org/html/rfc6587#section-3.4.1) is recommended as the
+message framing method. In a stream with [non-transparent framing](https://tools.ietf.org/html/rfc6587#section-3.4.2),
+promtail needs to wait for the next message to catch multi-line messages,
+therefore delays between messages can occur.
+
+You may need to increase the open files limit for the promtail process
+if many clients are connected. (`ulimit -Sn`)
+
+```yaml
+# TCP address to listen on. Has the format of "host:port".
+listen_address: <string>
+
+# The idle timeout for tcp syslog connections, default is 120 seconds.
+idle_timeout: <duration>
+
+# Whether to convert syslog structured data to labels.
+# A structured data entry of [example@99999 test="yes"] would become
+# the label "__syslog_message_sd_example_99999_test" with the value "yes".
+label_structured_data: <bool>
+
+# Label map to add to every log message.
+labels:
+  [ <labelname>: <labelvalue> ... ]
+```
+
+#### Syslog-NG Output Configuration
```

Comment:
I wonder if some of these examples should be moved (or copied) to `docs/clients/promtail/scraping.md`. The configuration document has mostly focused on the schema of the YAML file rather than the specifics of configuring it with other components (e.g., syslog-ng, rsyslog).  

What do you think?

Reply:
👍, I'll try to add a section in the style of the `scraping.md` documentation and link it.

Reply:
Take a look and tell me what you think :)

Reply:
I'm happy with that, nice job!